src/efftester.ml

```diff
@@ -1193,6 +1348,22 @@ module Shrinker = struct
     match m with
     | Lit _ -> m
     | Variable (t, z) -> if x = z then Variable (t, y) else m
+    | Constructor (typ, name, payload_lst) ->
+      let new_payload_lst = List.map (fun trm -> alpharename trm x y) payload_lst in
+      Constructor (typ, name, new_payload_lst)
+    | PatternMatch (typ, matched_trm, cases, eff) ->
+      let matched_trm' = alpharename matched_trm x y in
+      let rec alpharen_pat = function
+        | PattVar s when s = x -> PattVar y (* (?) Should we rename vars in patterns? *)
+        | PattVar s -> PattVar s
+        | PattConstr (typ, name, patt_lst) ->
+          let patt_lst' = List.map alpharen_pat patt_lst in
+          PattConstr (typ, name, patt_lst')
+      in
```

Comment:
According to the specific in the comment, only free occurrences of `x` should be modified, so in particular the branch `| Some x -> x` should remain unchanged. You probably need to use a function such as `occurs_in_pat x pat` that I mentioned in another comment above. (And you never need to traverse patterns to rename them, as the variables occurring within patterns are all binding occurrences, not free occurrences.)